src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -326,6 +353,42 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
             }
         }
 
+        private static void PopulateReferenceIdDictionary(IEnumerable<EntryComponent> bundleEntries, IDictionary<string, string> idDictionary)
+        {
+            foreach (EntryComponent entry in bundleEntries)
+            {
+                if (entry.Request.Method != HTTPVerb.POST)
+                {
+                    continue;
+                }
+
+                // We've already come across this ID
+                if (!idDictionary.TryGetValue(entry.FullUrl, out _))
+                {
```

Comment:
Since caller can put any value in fullURL, what will happen if  the same value is given for a FullURl for two different post entries?

Reply:
That would be stopped by the code in the PR that does the verification.